packages/adapter/lib/adapter/adapter.js

```diff
@@ -8731,6 +8734,53 @@ function Adapter(options) {
     this.pluginHandler = new PluginHandler(pluginSettings);
     this.pluginHandler.addPlugins(this.ioPack.common.plugins, [this.adapterDir, __dirname]); // first resolve from adapter directory, else from js-controller
 
+    /**
+     * This method returns the list of license that can be used by this adapter
+     * @param {boolean} all if return the licenses, that used by other instances (true) or only for this instance (false)
+     * @returns {object[]} list of suitable licenses
+    */
+    this.getSuitableLicenses = async all => {
+        const licenses = [];
+        try {
+            const obj = await this.getForeignObjectAsync('system.licenses');
+            if (obj && obj.native && obj.native.licenses && obj.native.licenses.length) {
+                const now = Date.now();
+                const cert = fs.readFileSync(__dirname + '/../../cert/cloudCert.crt');
+                const version = this.pack.version.split('.')[0];
+
+                obj.native.licenses.forEach(license => {
+                    try {
+                        const decoded = jwt.verify(license.json, cert);
+                        if (decoded.name && (!decoded.valid_till || license.valid_till === '0000-00-00 00:00:00' || new Date(license.valid_till).getTime() > now)) {
+                            if (decoded.name.startsWith('iobroker.' + this.name) && (all || !license.usedBy || license.usedBy === this.namespace)) {
+                                if (decoded.version === '<2' || decoded.version === '<1' || decoded.version === '<=1') {
+                                    if (version !== '0' && version !== '1') {
```

Comment:
can only adapters in version 0 or 1 have a license? some code comments would be good

Reply:
Good question ...

Reply:
or it should be docode.version and a typo?

Reply:
Added comments